src/core/trulens/apps/virtual.py

```diff
@@ -570,5 +576,56 @@ def add_record(
 
         return record
 
+    def add_dataframe(
+        self,
+        df,
+        feedback_mode: Optional[mod_feedback_schema.FeedbackMode] = None,
+    ) -> List[mod_record_schema.Record]:
+        """Add the given dataframe as records to the database and evaluate any pre-specified
+        feedbacks on them.
+
+        The class `VirtualRecord` may be useful for creating records for virtual models.
+
+        If `feedback_mode` is specified, will use that mode for these records only.
+        """
+
+        data_dict = df.to_dict("records")
+        records = []
+
+        for record in data_dict:
+            if "context" in record:
+                retriever = Select.RecordCalls.retriever
+                if retriever not in self.app:
+                    self.app[retriever] = None
+                self.app[retriever] = record["context"]
+
+                context_rets = (
+                    record["context"]
+                    if isinstance(record["context"], list)
+                    else [record["context"]]
+                )
+
+                rec = VirtualRecord(
+                    calls={
+                        retriever.get_context: dict(
+                            args=[record["prompt"]], rets=context_rets
+                        )
+                    },
+                    main_input=record["prompt"],
+                    main_output=record["response"],
+                )
+            else:
+                rec = VirtualRecord(
+                    main_input=record["prompt"],
```

Comment:
Keep this similar to ragas, question, contexts, response. I think

Reply:
We shouldn't break Virtual Record here by changing the name, but I updated the add_dataframe method to take question/answer/contexts.

Noting that this is a departure from the ground truth data naming, which requires `query` column name. This means if I want to use the same dataframe with some ground truths for ground truth evaluation, I'd need to rename the column for the ground truth calculation (not ideal).

Reply:
IMO we should go with `query`, `response`, `contexts`. Are you good with this @sfc-gh-pdharmana ?